modules/opsgenie-team/provider-datadog.tf

```diff
@@ -0,0 +1,15 @@
+// This is a custom provider-datadog.tf because it is always enabled, this is because we always need the datadog provider to be configured, even if the module is disabled.
+
+module "datadog_configuration" {
+  source  = "../datadog-configuration/modules/datadog_keys"
+  region  = var.region
+  enabled = true
```

Comment:
surprising gotcha. I guess this means that the provider will error out until datadog is configured?

Reply:
datadog provider needs to be available even if the opsgenie team is disabled, reason being is we need the datadog provider to delete the opsgenie integration for that team

Reply:
makes sense!